File: by-kotlin/kt-boj/solutions/1439-뒤집기.kt
import java.util.*
import kotlin.math.min

fun main() = with(Scanner(System.`in`)) {
    val data = next()

    fun solve(data: String): Int {
        val cnts = arrayListOf<Int>(0, 0)
        val cdata = data.toCharArray()
        if (cdata[0] == '1') {
            cnts[0] += 1
        } else {
            cnts[1] += 1
        }

        for (i in 0 until cdata.size - 1) {
            if (cdata[i] != cdata[i + 1]) {
                if (cdata[i + 1] == '1') {
                    cnts[0] += 1
                } else {
                    cnts[1] += 1
                }
            }
        }

        return min(cnts[0], cnts[1])
    }

    print(solve(data))
}

File: by-python/py-boj/1439-뒤집기.py
def solve(data):
    cnts = [0, 0]

    if data[0] == '0':
        cnts[1] += 1
    else:
        cnts[0] += 1

    for i in range(len(data) - 1):
        if data[i] != data[i + 1]:
            if data[i + 1] == '0':
                cnts[1] += 1
            else:
                cnts[0] += 1

    return min(cnts)


if __name__ == '__main__':
    data = input()
    print(solve(data))
